pkg/csi/service/vanilla/controller.go

```diff
@@ -926,39 +926,112 @@ func (c *controller) createFileVolume(ctx context.Context, req *csi.CreateVolume
 			"parsing storage class parameters failed with error: %+v", err)
 	}
 
-	var createVolumeSpec = common.CreateVolumeSpec{
-		CapacityMB: volSizeMB,
-		Name:       req.Name,
-		ScParams:   scParams,
-		VolumeType: common.FileVolumeType,
+	// Check if vCenter task for this volume is already registered as part of
+	// improved idempotency CR
+	log.Debugf("Checking if vCenter task for file volume %s is already registered.", req.Name)
+	var (
+		volTaskAlreadyRegistered bool
+		faultType                string
+		volumeID                 string
+		vcenter                  *cnsvsphere.VirtualCenter
+	)
+	// Get VirtualCenter instance
+	vcenter, err = c.manager.VcenterManager.GetVirtualCenter(ctx, c.manager.VcenterConfig.Host)
```

Comment:
Can we get the vCenter instance in the block where we need to create a task object, otherwise we are not making use of it.

Reply:
done